ansible/ansible-ai-connect-service
# AAP-17822 - mock WCA client and secret manager objects

Jira Issue: <https://issues.redhat.com/browse/AAP-17822>


## Description
    
- `WCA_CLIENT_BACKEND_TYPE`
- `WCA_SECRET_BACKEND_TYPE`
- `WCA_SECRET_MOCKER_SECRETS`
    
`WCA_SECRET_DUMMY_SECRETS` is a list of key:value with a , separator. key is the orgid, value is the secret.
When a secret with the string "valid", it means the backend will accept it has a valid string. e.g:
`WCA_SECRET_DUMMY_SECRETS=1009103:valid,11009104:not-valid`

## Testing

### Steps to test
1. Pull down the PR
2. Set environment variables similar to this example (FishShell):
```
set -x AUTHZ_BACKEND_TYPE "dummy"
set -x WCA_CLIENT_BACKEND_TYPE "dummy"
set -x WCA_SECRET_BACKEND_TYPE "dummy"

set -x AUTHZ_DUMMY_USERS_WITH_SEAT "gleboude1@redhat.com"
set -x AUTHZ_DUMMY_RH_ORG_ADMINS "gleboude1@redhat.com"
set -x AUTHZ_DUMMY_ORGS_WITH_SUBSCRIPTION "11009103"
set -x WCA_SECRET_DUMMY_SECRETS "11009103:valid,11009104:not-valid"
```
4. Ensure you've got the right information on the UI (e.g: no need to set the key)

## Production deployment
